src/amo/pages/UserProfile/index.js

```diff
@@ -112,6 +113,16 @@ export class UserProfileBase extends React.Component<InternalProps> {
         }),
       );
     }
+    let idOrNameCheck = { ...location}.pathname.split('/')[4];
```

Comment:
hu oh, what's that? Why not using `location` directly? In any case I don't think you need the `location`, there should be a `match` prop that does that (i.e. extracting URL parameters).